### DISCUSSION TITLE
Is there any way to limit request rate by user?

### DISCUSSION THREAD

**Discussion by nth347** [Kong/kong#11957] [Help]
Our users communicate with services through Kong, and we are exploring options to restrict the request rate on a per-user basis. I believe we should customize the [Rate Limiting](https://docs.konghq.com/hub/kong-inc/rate-limiting/) plugin to perform JWT inspection, extracting user credentials from the JWT to enforce the rate limit policies. How to do this actually?

However, I'm wondering if there are simpler methods to achieve this. Any suggestions?

**Comment by StarlightIbuki** [comment]
You could try setting "limit_by" to "credential" for a general user-based limiting.
Kong supports JWT introspecting with OIDC, an enterprise plugin. However, it's possible to implement a custom plugin to do the job.
